### PR TITLE
Add collections module

### LINKED ISSUE #21: managing collections basics
This means

- [x] create collection
- [x] delete collection 
- [ ] rename collection
- [ ] documentation
- [ ] tests

### LINKED ISSUE #32: use CARGO_TARGET_TMPDIR instead of /tmp for tests
The cargo doc specifies 
`CARGO_TARGET_TMPDIR — Only set when building [integration test](https://doc.rust-lang.org/cargo/reference/cargo-targets.html#integration-tests) or benchmark code. This is a path to a directory inside the target directory where integration tests or benchmarks are free to put any data needed by the tests/benches. Cargo initially creates this directory but doesn't manage its content in any way, this is the responsibility of the test code.`

It might be a good idea to use this instead of /tmp/ because it is not available on windows anyway.
Can be done at the same time of using std::path::Path in tests

### PR DESCRIPTION
**Pull request by ADBeveridge**

### CODE REVIEW

File: tests/basic_tests.rs [hunk 1]
```diff
@@ -11,7 +11,8 @@ mod tests
 
 	use rusqlite::{Connection};
 
-	static TEST_DIR: &str = "/tmp/";
+
+	static TEST_DIR: &str = env!("CARGO_TARGET_TMPDIR");
```
Comment (Sulfiore) [inline tests/basic_tests.rs:15]: I I suggest modifying all test paths to use a Path using .join() on this one to make the Windows test work

File: src/element.rs [hunk 2]
```diff
@@ -36,5 +36,14 @@ pub trait ElementDatabase
 /// Trait related to filesystem interactions
 pub trait ElementFilesystem
 {
-	fn insert_into(&self, fs: &Filesystem) -> Result<(), Error>;
+	// Trait for inserting an object from another part of of the filesystem.
+	#[allow(unused_variables)]
+	fn insert_into(&self, fs: &Filesystem) -> Result<(), Error> {
+		Ok(())
```
Comment (Sulfiore) [inline src/element.rs]: When working with traits it is possible to simply declare a function without implementing any behavior within it. You can keep what I written before and copy it.

File: src/ospl.rs [hunk 3]
```diff
@@ -44,9 +45,11 @@ use std::io::ErrorKind;
 use std::path::{Path, PathBuf};
 
 use database::Database;
+use element::ElementFilesystem;
```
Comment (Sulfiore) [inline src/ospl.rs]: This element should be removed as it serves no purpose outside of the context of the elements (Photo, Collections, etc).

File: src/collection.rs [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+#[allow(dead_code)]
+pub struct Collection
+{
+	pub id:					u32,
+	creation_datetime:		Option<NaiveDateTime>,
+	modification_datetime:	Option<NaiveDateTime>,
+	name:					String,
+	comment:				String,
+
+	path_on_fs:				PathBuf
```
Comment (Sulfiore) [inline src/collection.rs]: I believe that the variable (path_on_fs) in question is unnecessary here as it is only used to track imported folders that have not been copied, and this scenario will not occur.

File: src/element.rs (hunk 2 above)
Comment (ADBeveridge) [reply src/element.rs]: I'm pretty sure I resolved this as you wanted.

**Comment by Sulfiore** [comment]
It seems like some of you code doesn't belong to some commit can you fix that please ?

**Comment by ADBeveridge** [comment]
Ok, I split the implementation of the fs object into its own commit.

**Pull request closed without merging**